File: Module/ALittleIDE/Other/GameLibrary/Core/JSScript/Core/Net/HttpSender.js
{
if (typeof ALittle === "undefined") ALittle = {};


ALittle.IHttpSender = JavaScript.Class(undefined, {
	HandleSucceed : function() {
	},
	HandleFailed : function(reason) {
	},
	SendRPC : function(thread, method, content) {
		throw new Error("not impl");
	},
	Invoke : function(method, client, content) {
		return new Promise(function(___COROUTINE, ___) {
			if (___COROUTINE === undefined) {
				___COROUTINE(["当前不是协程", undefined]); return;
			}
			client.SendRPC(___COROUTINE, method, content);
			return;
			___COROUTINE();
		});
	},
}, "ALittle.IHttpSender");

}

File: Module/ALittleIDE/Other/GameLibrary/Core/Script/Core/Net/HttpSender.lua

module("ALittle", package.seeall)

local ___pairs = pairs
local ___ipairs = ipairs


IHttpSender = Lua.Class(nil, "ALittle.IHttpSender")

function IHttpSender:HandleSucceed()
end

function IHttpSender:HandleFailed(reason)
end

function IHttpSender:SendRPC(thread, method, content)
	Lua.Throw("not impl")
end

function IHttpSender.Invoke(method, client, content)
local ___COROUTINE = coroutine.running()
	if ___COROUTINE == nil then
		return "当前不是协程", nil
	end
	client:SendRPC(___COROUTINE, method, content)
	return coroutine.yield()
end

